vktrace/vktrace_layer/vktrace_lib_trace.cpp

```diff
@@ -1824,6 +1824,164 @@ VKTRACER_EXPORT VKAPI_ATTR VkResult VKAPI_CALL __HOOKED_vkAllocateDescriptorSets
     return result;
 }
 
+// When define DescriptorSet Layout, the binding number is also defined. by Doc,
+// the descriptor bindings can be specified sparsely so that not all binding
+// numbers between 0 and the maximum binding number. the function is used to
+// convert the binding number to binding index starting from 0.
+uint32_t get_binding_index(VkDescriptorSet dstSet, uint32_t binding) {
+    uint32_t binding_index = 0;
+    trim::ObjectInfo* pInfo = trim::get_DescriptorSet_objectInfo(dstSet);
+    for (uint32_t i = 0; i < pInfo->ObjectInfo.DescriptorSet.numBindings; i++) {
+        if (binding == pInfo->ObjectInfo.DescriptorSet.pWriteDescriptorSets[i].dstBinding) {
+            binding_index = i;
+            break;
+        }
+    }
+    return binding_index;
+}
```

Comment:
This function will return 0 if binding is not found in the descriptor set. 0 can also be a valid return value when binding is found. Is this what you want?

Reply:
yes, that mean the target app has error or something wrong with trim descriptorset tracking. now will add a commit to add assert fail and let trim ignore the binding update for release build.